### PR TITLE
Added support for Puli FilesystemRepository

### PR DESCRIPTION
**Pull request by dantleech**

### CODE REVIEW

**Comment by lsmith77** [comment]
that looks rather easy :)

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -48,6 +48,16 @@ public function getConfigTreeBuilder()
                                 ->end()
                             ->end()
                         ->end()
+                        ->arrayNode('filesystem')
```
Comment (WouterJ) [inline DependencyInjection/Configuration.php]: this does not support XML, as you are missing a `->fixXmlConfig` method

Comment (WouterJ) [inline DependencyInjection/Configuration.php]: I'm not sure I like the way you fixed it with the other options btw, as it currently means that XML would look like:

````xml
<repository>
    <doctrine-phpcr-odm-repository basepath="..." />
</repository>
````

Comment (WouterJ) [inline DependencyInjection/Configuration.php]: Maybe something like the RoutingAuto config is better

Comment (dantleech) [inline DependencyInjection/Configuration.php]: Yeah

````
repositories:
    my_repository:
        type: doctrine-phpcr-odm
        options:
            # ...
````

Would be prefered, but then we lose configuration introspection as I think we would have to simply define an array prototype for the options..

But maybe we should change it is not the ideal configuration structure.

**Pull request merged**